src/dynamicprompts/samplers/utils.py

```diff
@@ -50,3 +54,48 @@ def get_wildcard_not_found_fallback(
     res = SamplingResult(text=wrapped_wildcard)
     while True:
         yield res
+
+def replace_wildcard_variables(
+    command: WildcardCommand, 
+    *, 
+    context: SamplingContext,
+) -> WildcardCommand:    
```

Comment:
For performance, this should probably have an early-return condition, so if there are no variables in the current context, or there's no way the wildcard text can contain a variable command (`'$' not in command.wildcard`, pretty much 😁), we should just return the original command without bothering with the rest.

Reply:
Added a test to return the original command if it does not include the wildcard start characters.